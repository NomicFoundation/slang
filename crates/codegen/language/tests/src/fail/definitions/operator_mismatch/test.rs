#![allow(unused_crate_dependencies)]

codegen_language_macros::compile!(Language(
    name = Foo,
    documentation_dir = "foo/bar",
    root_item = Bar,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [Topic(
            title = "Topic One",
            items = [
                Precedence(
                    name = Expression,
                    precedence_expressions = [PrecedenceExpression(
                        name = Foo,
                        operators = [
                            PrecedenceOperator(
                                model = BinaryLeftAssociative,
                                fields = (operator = Required(Baz1))
                            ),
                            PrecedenceOperator(
                                model = Prefix,
                                fields = (operator = Required(Baz1))
                            )
                        ]
                    )],
                    primary_expressions = [PrimaryExpression(reference = Baz1)]
                ),
                Token(
                    name = Baz1,
                    definitions = [TokenDefinition(scanner = Atom("baz1"))]
                )
            ]
        )]
    )]
));

fn main() {}
