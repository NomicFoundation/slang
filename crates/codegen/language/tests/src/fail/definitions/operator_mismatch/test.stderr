error: All operators under the same expression must have the same model and type.
  --> src/fail/definitions/operator_mismatch/test.rs:18:32
   |
18 |                         name = Foo,
   |                                ^^^
